public/app/features/explore/utils/set.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Performs a shallow comparison of two sets with the same item type.
+ */
+export function equal<T>(a: Set<T>, b: Set<T>): boolean {
+  if (a.size !== b.size) {
+    return false;
+  }
+  const it = a.values();
+  while (true) {
+    const { value, done } = it.next();
+    if (b.has(value)) {
+      return false;
```

Comment:
Can we have a test for util functions? I'm tripping over this line.

Reply:
Ended up being good catch—tests revealed I'd implemented the operations incorrectly.